### PR TITLE
ceph_fuse: pid_file default to empty

### PR DESCRIPTION
**Pull request by smithfarm**
Fixes: http://tracker.ceph.com/issues/18309
Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
For reference: 

* https://github.com/ceph/ceph/blob/master/src/common/common_init.cc#L84-L87
* https://github.com/ceph/ceph/blob/master/src/global/pidfile.cc#L213-L216

File: src/ceph_fuse.cc [hunk 1]
```diff
@@ -78,7 +78,10 @@ int main(int argc, const char **argv, const char *envp[]) {
   }
   env_to_vec(args);
 
-  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT,
+  std::vector<const char*> def_args;
+  def_args.push_back("--pid-file="); // pid_file default to empty
```
Comment (tchaikov) [inline src/ceph_fuse.cc]: nit, could just put
```c++
std::vector<const char*> def_args{"--pid-file="};
```

**Comment by smithfarm** [comment]
This should make `fs/recovery/{begin.yaml clusters/4-remote-clients.yaml fs/xfs.yaml mount/fuse.yaml objectstore/filestore.yaml overrides/{debug.yaml frag_enable.yaml fuse-unmounted.yaml whitelist_wrongly_marked_down.yaml} tasks/volume-client.yaml}` succeed.

**Comment by smithfarm** [comment]
Changelog:

* standardize on `std::vector` even though there is a `using namespace std` at the top of the file
* set def_args via `{}` as suggested by @tchaikov
* add commit to remove dead code following `usage()` call

**Comment by smithfarm** [comment]
`teuthology-suite -s fs -c wip-18309-alt --filter "recovery/{begin.yaml clusters/4-remote-clients.yaml fs/xfs.yaml mount/fuse.yaml objectstore/filestore.yaml overrides/{debug.yaml frag_enable.yaml fuse-unmounted.yaml whitelist_wrongly_marked_down.yaml} tasks/volume-client.yaml}"`

**waiting for build**

**Comment by smithfarm** [comment]
Both tests appear to fail because of the pidfile issue.

**Comment by jcsp** [comment]
@smithfarm weird.  I had a quick look through arg parsing code and I can't see why the --pidfile= is being ignored.

**Comment by jcsp** [comment]
Ah, this is failing because the teuthology ceph.conf.template has `pid file = /var/run/ceph/$cluster-$name.pid` in `[global]`

**Comment by smithfarm** [comment]
So we need to add something like

```
overrides:
  ceph:
    conf:
      global:
        pid file = ""
```

to the test definition?

**Comment by smithfarm** [comment]
(I suspected that pid file name was coming from the test, but I grepping for it in `qa/` turned up nothing, now I know why)

**Comment by smithfarm** [comment]
@jcsp Something like in the commit I just added?

**Comment by jcsp** [comment]
@smithfarm yeah, but we probably want to set the override for `client` rather than `global` to avoid collaterally disabling pidfiles for daemons in the tests.

**Comment by jcsp** [comment]
This would still cause issues for administrators if they had set their own global pid file setting the way teuthology does, but I think I'm okay with that -- the alternative would be squashing any existing pid_file setting from ceph_fuse, which would remove the admin's ability to optionally have the fuse client generate a pid file (not sure why they'd want to, but doesn't seem completely inconceivable)

**Comment by smithfarm** [comment]
@jcsp The test in question has four clients, `client.[0123]`, but I'm guessing that

```
overrides:
  ceph:
    conf:
      client:
        pid file: ""
```

will cover them all.

**Comment by smithfarm** [comment]
Hm, the `no-client-pidfile.yaml` is so general, maybe `qa/overrides` is a better place for it.

**Comment by smithfarm** [comment]
`teuthology-suite -s fs -c wip-18309-alt --filter "recovery/{begin.yaml clusters/4-remote-clients.yaml fs/xfs.yaml mount/fuse.yaml objectstore/filestore.yaml overrides/{debug.yaml frag_enable.yaml fuse-unmounted.yaml whitelist_wrongly_marked_down.yaml} tasks/volume-client.yaml}" --priority 101 --email ncutler@suse.cz -m smithi` (with patch - should succeed)

**pending build**

**Comment by smithfarm** [comment]
`teuthology-suite -s fs -c wip-18309-alt --filter "recovery/{begin.yaml clusters/4-remote-clients.yaml fs/xfs.yaml mount/fuse.yaml objectstore/filestore.yaml overrides/{debug.yaml frag_enable.yaml fuse-unmounted.yaml no_client_pidfile.yaml whitelist_wrongly_marked_down.yaml} tasks/volume-client.yaml}" --priority 101 --email ncutler@suse.cz -m smithi` (with patch - should succeed)

**pending build**

**Comment by tchaikov** [comment]
retest this please 
```
Submodule path 'src/erasure-code/jerasure/gf-complete': checked out 'd61b4d377a6a3f202de8a3541abf6383313adab7'
Cloning into 'src/erasure-code/jerasure/jerasure'...
fatal: unable to access 'https://github.com/ceph/jerasure.git/': Could not resolve host: github.com
Clone of 'https://github.com/ceph/jerasure.git' into submodule path 'src/erasure-code/jerasure/jerasure' failed
...
-- Performing Test HAVE_INTEL_SSE4_2 - Success
CMake Error at src/CMakeLists.txt:568 (add_subdirectory):
  The source directory

    /home/jenkins-build/build/workspace/ceph-pull-requests/src/lua

  does not contain a CMakeLists.txt file.
```

seems the submodules are not fully sync'ed due to networking issues.

**Comment by tchaikov** [comment]
@smithfarm scheduled at http://pulpito.front.sepia.ceph.com:80/kchai-2017-01-06_12:31:35-fs-wip-18309-alt---basic-mira/

**Comment by smithfarm** [comment]
@tchaikov Thanks! If you and @jcsp are satisfied with the last two commits, I guess it's good to merge.

**Comment by jcsp** [comment]
:+1: thanks @smithfarm

**Pull request merged**